File: leet169/leet169.py
#https://leetcode.com/problems/majority-element/description/
from typing import List

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 1
        majority = nums[0]
        for i in nums:
            if i == majority:
                count += 1
            else:
                count -= 1
                if count == 0:
                    majority = i
                    count = 1
        return majority

File: leet169/leet169.cpp
//https://leetcode.com/problems/majority-element/description/
#include <vector>

class Solution {
public:
    int majorityElement(std::vector<int>& nums) {
        int majority = nums[0];
        int count = 1;
        for(int i = 1; i < nums.size(); i++){
            if(nums[i] == majority){
                count++;
            }
            else{
                count--;
                if(count == 0){
                    majority = nums[i];
                    count = 1;
                }
            }
        }
        return majority;
    }
};